src/SonarScanner.MSBuild.Common/Interfaces/IOperatingSystemProvider.cs

```diff
@@ -35,6 +35,8 @@ public interface IOperatingSystemProvider
 {
     public PlatformOS OperatingSystem();
 
+    public bool IsUnix();
```

Comment:
I would add a doc comment, so the caller knows about MacOS
```suggestion
    /// <summary>Returns <see langword="true" /> for all *nix-like operating systems, including Linux, Alpine, and MacOS.</summary>
    public bool IsUnix();
```